lightning/src/ln/channelmanager.rs

```diff
@@ -3490,13 +3532,19 @@ where
 				err: format!("Payment with intercept id {} not found", log_bytes!(intercept_id.0))
 			})?;
 
+		let skimmed_fee_msat =
+			payment.forward_info.outgoing_amt_msat.checked_sub(amt_to_forward_msat)
+			.ok_or_else(|| APIError::APIMisuseError {
+				err: format!("Attempted to forward intercept with id {} with amount {} msat, maximum forward amount was {} msat", log_bytes!(intercept_id.0), amt_to_forward_msat, payment.forward_info.outgoing_amt_msat)
+			})?;
```

Comment:
The docs for `forward_intercepted_htlcs` and `Event::HTLCIntercepted::expected_outbound_amount_msat` say we don't check whether they forward more than they received, which is technically still true, but should probably be updated now that we check they forward less than they were intended to forward (unless negative fees were a thing I guess)?

Also I think it might be helpful to include some docs on either `forward_intercepted_htlcs` or `Event::HTLCIntercepted` to make it more explicit that users have the option to send less than was expected to take a skimmed fee?

Reply:
Updated some intercept docs. Yeah I don't think this error makes sense, continued the conversation here: https://github.com/lightningdevkit/rust-lightning/pull/2319#discussion_r1235518032